streaming/vision/convert/coco.py

```diff
@@ -0,0 +1,231 @@
+# Copyright 2022 MosaicML Composer authors
+
+"""COCO 2017 streaming dataset conversion scripts."""
+
+import json
+import os
+from argparse import ArgumentParser, Namespace
+from typing import Dict, Iterable
+
+import numpy as np
+import torch
+from PIL import Image
+from torch.utils.data import Dataset
+from tqdm import tqdm
+
+from streaming.base import MDSWriter
+from streaming.vision.convert.base import get_list_arg
+
+
+class _COCODetection(Dataset):
+    """PyTorch Dataset for the COCO dataset.
+
+    Args:
+        img_folder (str): Path to a COCO folder.
+        annotate_file (str): Path to a file that contains image id, annotations (e.g., bounding boxes and object
+            classes) etc.
+    """
+
+    def __init__(self, img_folder: str, annotate_file: str):
+        self.img_folder = img_folder
+        self.annotate_file = annotate_file
+
+        # Start processing annotation
+        with open(annotate_file) as fin:
+            self.data = json.load(fin)
+
+        self.images = {}
+
+        self.label_map = {}
+        self.label_info = {}
+        # 0 stands for the background
+        cnt = 0
+        self.label_info[cnt] = 'background'
+        for cat in self.data['categories']:
+            cnt += 1
+            self.label_map[cat['id']] = cnt
+            self.label_info[cnt] = cat['name']
+
+        # build inference for images
+        for img in self.data['images']:
+            img_id = img['id']
+            img_name = img['file_name']
+            img_size = (img['height'], img['width'])
+            if img_id in self.images:
+                raise Exception('dulpicated image record')
```

Comment:
spelling

Reply:
Fixed in https://github.com/mosaicml/streaming/pull/5/commits/e1eaca7428674cb2941960d1ad7c39c7091c1b12.